### ISSUE TITLE
"mev_simBundle" rpc method for enhanced transaction simulation

### ISSUE THREAD

**Issue by zently** [paradigmxyz/reth#9472]
### Describe the feature

`mev_simBundle` is a rpc endpoint spec used for simulating a bundle in the virtual environment.
First introduced by the flashbot, it is widely recognized as **THE BEST** way to simulate a bundle (possibly multiple tx)

Currently `eth_callBundle` works well for similar purpose but `mev_simBundle` is better in that,
it returns transaction logs of the simulated transaction.

Having transaction logs is critical in multiple occasions such as,

 - Pre-validating transaction before sending real transaction (for protection, test)
 - Test / debugging contract
 - Searchers checking whether specific pending transaction yields specific action (e.g. token transfer, swap)

Currently, within "mev.rs" struct for the `mev_simBundle` is already defined fully, 
while its implementation is missing in the "bundle.rs".

### Additional context

[Flashbots document](https://docs.flashbots.net/flashbots-mev-share/searchers/debugging#simulate-your-bundle)

**Comment by mattsse** [comment]
not opposed to this because I see this being useful and only does simulation
I think the reference impl is

https://github.com/flashbots/builder/blob/df9c765067d57ab4b2d0ad39dbb156cbe4965778/internal/ethapi/sbundle_api.go#L205

https://github.com/flashbots/builder/blob/df9c765067d57ab4b2d0ad39dbb156cbe4965778/core/sbundle_sim.go#L58



## TODO
* add new mev namespace
* port simBundle function

**Comment by kien6034** [comment]
hello, can i work on this issue?

**Comment by mattsse** [comment]
yes please,

for ref ptal at the call_bundle impl:

https://github.com/paradigmxyz/reth/blob/df9cb81ea61a1393b1caf5da50e43d633920f7da/crates/rpc/rpc/src/eth/bundle.rs#L51-L51

**Comment by kien6034** [comment]
@mattsse  
thank you sir

**Comment by jekkos** [comment]
Any progress on this? It would be nice to have the tx logs in the callbundle result. This used to be possible in a patched version of mev-geth as well.

**Comment by SkandaBhat** [comment]
@mattsse if there have been no updates on this, I would like to pick this up. Could you please assign this to me?

**Comment by SkandaBhat** [comment]
Working on the impl, will share a draft soon!

**Comment by SkandaBhat** [comment]
@mattsse We need to implement `Bundle` inside `BundleItem` in https://github.com/alloy-rs/alloy/blob/8d99c28da17232a26afe7af0251fde2cd00a89fe/crates/rpc-types-mev/src/mev_calls.rs#L83. I can implement and raise a PR over there

**Comment by mattsse** [comment]
just checking in @SkandaBhat 
types should now be all unblocked, are you on this or should I open this for others?

**Comment by SkandaBhat** [comment]
working on this, should have a draft soon @mattsse

**Comment by mattsse** [comment]
feel free to open a draft pr early

**Comment by SkandaBhat** [comment]
Will create a draft today @mattsse
